### ISSUE TITLE
BGCurve line render component osx / ios issue 

### ISSUE THREAD

**Issue by antonlind** [bansheeGz/BGCurve#4]
Hi!
I've been using BG curve for a new project & overall I must say it is great :).
However, I've found an issue with the line render component on osx/ios. I have created a path, constricted to xz, a road basically. On Windows/Android the vertices faces up, which seems correct. On OSX and when built for IOS the vertices faces right. I can counter this by rotating the entire GameObject around z on osx, but of course all the control-points get the wrong orientation then. This issue is not present when I replicate the road with the Unity line renderer only. So there seems to be some transformation issue somewhere.

**Comment by banshee-gzzz** [comment]
Hi,
Could you elaborate what you mean by "orientation"?
LineRenderer operates with a set of positions, there is no any orientation.
And under the hood, we still use Unity's standard LineRenderer- we fill its point positions and it's the only field we are changing.

**Comment by antonlind** [comment]
By orientation I mean how the segments of the road / line are rotated. On windows the surface is pointing up, ↑ . On OSX the surface is poiting right → . I know there is no rotation of the points in the linerender, however it is possible to rotate the entire gameobject, and then the facing changes. Do you touch the gameobject transform in any way? It's like there is an offset of 90 degrees, but it not reflected in the inspector. It just says 0 rotation there.
In my opinion the linerenderer is somewhay buggy in unity anyways, but since the issue does not appear without the use of the BGCurve component I thought it was worth reporting

**Comment by banshee-gzzz** [comment]
if you rotate the GameObject with BGCurve (or any of its parents), all points positions will be rotated as well. 

However, I do not think there is any "bug" with this rotation, simply because we do not have any platform-dependent code, which would behave differently on different platforms.
There is also an option to disable or delete BGCcLineRenderer component after you constructed your initial LineRenderer component (before rotation)

If you still think it's an issue with BGCurve, could send us a simple example scene, which would illustrate your issue, so I could look into it?

**Comment by antonlind** [comment]
Here is an example project:
https://drive.google.com/drive/folders/0BzEHDm7pQQkSbm9taVd5S2pQZDQ?usp=sharing

There are also two screenshots to illustrate the difference between osx & windows with the exact same setup. If you feel that there are no transformations involved on your end that can cause this, then I think It's most likely some issue with Unity's line render implementation and that your plugin just triggers this issue somehow. There are probably several ways for me to correct it, with a script or shader, so I will look into doing that if there is no fix to be done on your end.

**Comment by banshee-gzzz** [comment]
Thanks for the example, I got what you mean.
No, we certainly can not fix it, cause we do not control it. All we do is setting line's positions.
Unity's LineRenderer has only positions, it does not have any normal vectors to define orientation, and I have no idea how Unity guys define their "width" direction, it actually can be any direction.
I noticed you had to move the whole "road" 100 meters up to get proper width direction.

However, if you can wait for (around) the end of this week, we could try to provide you with a standard component which would do what you want using standard Unity's Mesh, e.g. it will sweep a line along the curve with defined normal direction (e.g. "up" in your case).
Let me know if you're interested.

**Comment by antonlind** [comment]
I am definitely interested, it would be great! I get by with having it working correctly on windows / android for now, so a week or even two is not a problem atm.

**Comment by banshee-gzzz** [comment]
ok, I will write here as soon as it's ready (probably in the beginning of the next week)

**Comment by banshee-gzzz** [comment]
We have added v.1.2.3 beta https://github.com/bansheeGz/BGCurve/blob/master/Package/BGCurve1.2.3.beta.unitypackage  with BGCcSweep2D component.

BGCurveTestComponents.unity  scene has an example.
This new component has an issue though- with sharp angles generated polygons can overlap. This issue can not be easily fixed, so we'll leave it as it is.

Please, test it and let us know if it worked for you.

**Comment by antonlind** [comment]
Hi!
That was quick:) I have tested the new component briefly and it seems to be working well. I will keep it in my game/project and report any new issues that I might find when I run into them.
I did find two small things, but nothing mayor:
- When the curve is set to 2d mode and the gameobject is moved around the mesh does not line up properly with the curve, this can be demonstrated by moving the curve in the testscene. However if set to 3d mode this problem goes away, but I get a warning saying that the curve should be in 2d mode.
- The mesh filter of the render is exposed, so if any mesh is added there it will be overwritten. This is not really a problem at all since that is not how to use this component, but maybe that field should be hidden. 

Again, many thanks for this and I will get back to you if I have any more feedback after extended use!
